### PR TITLE
feat: add `unsafe` expression

### PR DESCRIPTION
**Pull request by magnus-madsen**
This is a shorthand for an unchecked effect cast to pure. 

We will want it for our OO-refactoring, as shown with one example in String.flix

We can discuss later if it should have richer semantics.

### CODE REVIEW

**Comment by JonathanStarup** [comment]
I know you said later, but it seems likely (when wrapping mutable things in java fx) that would want the cast to some region `r` instead of `{}`. what about `unsafe <effect> { <exp> }`

File: main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala [hunk 1]
```diff
@@ -1594,6 +1595,17 @@ object Weeder2 {
       }
     }
 
+    private def visitUnsafeExpr(tree: Tree): Validation[Expr, CompilationMessage] = {
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala:1598]: Does this belong in Weeder or Desugarer?

File: main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala [hunk 2]
```diff
@@ -1594,6 +1595,17 @@ object Weeder2 {
...
+      expect(tree, TreeKind.Expr.Unsafe)
+      mapN(pickExpr(tree)) {
+        expr =>
+          // We desugar an unsafe expression to an unchecked cast to pure.
+          val declaredType = None
+          val declaredEff = Some(Type.Pure(tree.loc))
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala:1604]: Synthetic location

File: main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala (hunk 1 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala:1598]: Fixed

File: main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala (hunk 2 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala]: Fixed.

**Comment by magnus-madsen** [comment]
> I know you said later, but it seems likely (when wrapping mutable things in java fx) that would want the cast to some region `r` instead of `{}`. what about `unsafe <effect> { <exp> }`

I think these are all good points, but I am just adding it now - as is - so Chanattan can start refactoring the library.

**Comment by magnus-madsen** [comment]
@mlutze RFR

**Review by mlutze — APPROVED** [review]

**Pull request merged**